### PR TITLE
Fix GCHPctmEnv small numerical bugs

### LINKED ISSUE #445: GCHP mass flux wrong moisture correction
### Your name

Yuanjian Zhang

### Your affiliation

WashU

### What happened? What did you expect to happen?

https://github.com/geoschem/GCHP/blob/e499969f3831261e1fd15774b9fc68d01d012ba2/src/GCHP_GridComp/GCHPctmEnv_GridComp/GCHPctmEnv_GridCompMod.F90#L873-L885

Humidity is positive so `1-SPHU` is smaller than 1, so `dryMF = MF/(1-SPHU)` is bigger than `MF`, which does not make sense to me. Shouldn't mass flux from GMAO archives be multiplied instead of being divided by `1-SPHU`?

### LINKED ISSUE #455: Opposite sign of UpwardsMassflux diagnostic
### Your name

Yuanjian Zhang

### Your affiliation

WashU

### What happened? What did you expect to happen?

UpwardsMassflux currently is positive = "down" instead of intuitively positive = "up".

### Evidence from diagnostics
image

UpwardsMassflux diagnostic from a Jan 2019 transport tracer run shows negative values around the equator, which, according to Hadley Circulation, should be upwards. 

### Evidence from codes
We can derive the mass balance below:

$$Tendency\\_of\\_Mass\\_Above\\_Level\\_K = BK(k)\times PIT = CONV(k) + UPWARDS\\_MFZ(k)$$

where `UPWARDS_MFZ` is the UPWARDS mass flux, `CONV` is accumulative horizontal mass flux convergence from top to level k, `BK` is the hybrid parameter b, `PIT` is the surface pressure tendency.

[FVdycoreCubed_GridComp/FV_StateMod.F90](https://github.com/geoschem/FVdycoreCubed_GridComp/blob/af424623fd01be0e640a655414e7709f6fc45f6a/FV_StateMod.F90#L3168-L3203)
```fortran
! Compute the vertical mass flux
!
!   Compute Convergence of the horizontal Mass flux
    do k=1,npz
       do j=jsc,jec
          do i=isc,iec
             conv(i,j,k) = ( xfx(i,j,k) - xfx(i+1,j,k) +  &
                             yfx(i,j,k) - yfx(i,j+1,k) )
          enddo
       enddo
    enddo
!   Surface pressure tendency
    pit(:,:) = 0.0
    do k=1,npz
       do j=jsc,jec
          do i=isc,iec
             pit(i,j) = pit(i,j) + conv(i,j,k)
          enddo
       enddo
    enddo
!   Sum over levels
    do k=2,npz
       do j=jsc,jec
          do i=isc,iec
             conv(i,j,k) = conv(i,j,k) + conv(i,j,k-1)
          enddo
       enddo
    enddo
    mfz(:,:,:) = 0.0
    do k=2,npz
       do j=jsc,jec
          do i=isc,iec
             mfz(i,j,k) = ( conv(i,j,k-1)  - FV_Atm(1)%bk(k)*pit(i,j) )/(MAPL_GRAV*fv_atm(1)%gridstruct%area(i,j))  ! Kg/m^2/s
          enddo
       enddo
    enddo
```

In the codes above, however, `mfz(k) = (conv(k) - bk(k)*pit)*unit_conv`, proving mfz here is `-upwards_mfz`, in other words, downwards_mfz.

https://github.com/geoschem/GCHP/blob/e499969f3831261e1fd15774b9fc68d01d012ba2/src/GCHP_GridComp/GCHPctmEnv_GridComp/GCHPctmEnv_GridCompMod.F90#L956-L957

Here the flipping only converts the GMAO vertical conventions to GEOS-Chem vertical conventions, namely from level 1 being top to level 1 being surface.

### PR DESCRIPTION
**Pull request by yuanjianz**
### Name and Institution (Required)

Name: Yuanjian Zhang
Institution: WashU

### Describe the update

1. Correct dry mass flux derivation.
2. Correct upwardsMassflux sign.

### Expected changes

1. Dry mass flux will be slightly smaller than before but is expected.
2. UpwardsMassflux diagnostic will correctly be positive = "up"

### Related Github Issue

closes #445
closes #455

### CODE REVIEW

**Comment by lizziel** [comment]
Update #1 affects only GCHP runs that import mass fluxes from offline meteorology which makes this a zero diff update for benchmarking. I will therefore put milestone 14.5.1.

File: src/GCHP_GridComp/GCHPctmEnv_GridComp/GCHPctmEnv_GridCompMod.F90 [hunk 1]
```diff
@@ -953,8 +953,9 @@ subroutine prepare_massflux_exports(IMPORT, EXPORT, PLE, dt, RC)
          ! Get vertical mass flux
          call fv_getVerticalMassFlux(MFX_EXPORT, MFY_EXPORT, UpwardsMassFlux, dt)
 
-         ! Flip vertical so that GCHP diagnostic is positive="up"
-         UpwardsMassFlux(:,:,:) = UpwardsMassFlux(:,:,LM:0:-1)/dt
+         ! Flip vertical so that GCHP diagnostic level is follwing GEOS-Chem convetion
```
Comment (lizziel) [inline src/GCHP_GridComp/GCHPctmEnv_GridComp/GCHPctmEnv_GridCompMod.F90]: Typo in 'following'

**Review by lizziel — CHANGES_REQUESTED** [review]
Please also update the changelog with the updates.

File: src/GCHP_GridComp/GCHPctmEnv_GridComp/GCHPctmEnv_GridCompMod.F90 (hunk 1 above)
Comment (yuanjianz) [reply src/GCHP_GridComp/GCHPctmEnv_GridComp/GCHPctmEnv_GridCompMod.F90]: Typo fixed and ChangeLog Updated

**Review by lizziel — APPROVED** [review]

**Comment by lizziel** [comment]
@yuanjianz, I am rebasing to avoid merge conflicts and streamline the git history when I bring this into dev/no-diff-to-benchmark.

**Pull request merged**